### PR TITLE
US15_Session_Visualization

### PR DESCRIPTION
**Pull request by saeie07**
Closing #102, #103

### CODE REVIEW

File: client/src/views/HomeView.vue [hunk 1]
```diff
@@ -1,52 +1,295 @@
...
+onMounted(() => {
+  const container = document.querySelector('.visualization-container');
+  if (container) {
+    containerWidth = container.clientWidth;
+    containerHeight = container.clientHeight;
+  }
+});
+
+// Assign positions randomly
+const generateRandomGridPositions = (flowerCount: number) => {
```
Comment (danielamormocea) [inline client/src/views/HomeView.vue:157]: Could you please move this and the function below into a separate file?  Someting like flowerHelpers? I think it would make it easier to read this file.

File: client/src/views/HomeView.vue [hunk 2]
```diff
@@ -1,52 +1,295 @@
...
-    <div class="previously-played">
-
+      <div class="info-box" :class="{ active: infoVisible }" @click="toggleInfo">
+        <div> i </div>
+      </div>
+      <div class="zoom-controls">
+        <button @click="zoomIn">+</button>
+        <button @click="zoomOut">-</button>
+      </div>
+      <div :class="['visualization-container']" >
```
Comment (danielamormocea) [inline client/src/views/HomeView.vue:269]: I think you can just use 'class="visualization-container"' without the array brackets.

**Review by danielamormocea — CHANGES_REQUESTED** [review]
Could you also please add a screenshot / screencast of how it looks?

File: client/src/views/HomeView.vue (hunk 2 above)
Comment (saeie07) [reply client/src/views/HomeView.vue:269]: done, thought it was necessary because of the arrays but adapted.

File: client/src/views/HomeView.vue (hunk 1 above)
Comment (saeie07) [reply client/src/views/HomeView.vue:157]: do you think this is necessary for other reasons than readability?
because I was thinking that we only need these functions in Home and will not call it anywhere else, so is a separate file necessary? Otherwise, we would have to do it with everything, also the zoom functions. 

also, where would you add this separate helper file?

**Comment by saeie07** [comment]
Screenshot 2024-08-21 at 11 56 08
Screenshot 2024-08-21 at 11 56 30

File: client/src/views/HomeView.vue (hunk 1 above)
Comment (danielamormocea) [reply client/src/views/HomeView.vue:157]: It's true they they'll only be used here, but separating them could improve maintainability. I see it as: homeview has the general structure, and then we have other files with specific logic to them. Otherwise, I worry this file will grow increasingly larger and harder to maintain / read (I see it is already quite long). I also have trouble understanding the logic of these parts, so maybe separating them would increase understandability (comments would also be nice, but I don't think I'll make a changes in here, as I don't fully understand it, so if you're the only one dealing with this, that's fine, or maybe you still need to change things, so comments can come later). 

Basically if these are being moved, it's easier for anyone, I think, to understand the homeview structure, without the added complexity of these computational functions. 
 
As for location, we could have a directory under src, maybe named "visualization"/ "helpers" (totally up to you), and have the file in there.
